### PR TITLE
Update utils.py - Fixed save_checkpoint with scheduler set to None

### PR DESCRIPTION
**Pull request by peacefulotter**
The `save_checkpoint` raised an error if scheduler was set to None. This is fixed by setting the scheduler state_dict to None in case the scheduler is not set.

### CODE REVIEW

**Comment by mkrima** [comment]
LGTM

File: src/optim/base.py [hunk 1]
```diff
@@ -40,7 +40,8 @@ def train_base(model, opt, data, scheduler, iterations, acc_steps, batch_size, s
         if extra_args.grad_clip != 0.0:
             torch.nn.utils.clip_grad_norm_(model.parameters(), extra_args.grad_clip)
         opt.step()
-        scheduler.step()
+        if scheduler != None:
```
Comment (mkrima) [inline src/optim/base.py:43]: nit: if scheduler is not None

(otherwise LGTM)

Comment (Andron00e) [reply src/optim/base.py:43]: hello, this kind of fix was already proposed in the [soap](https://github.com/epfml/llm-baselines/tree/soap) branch,
but we didn't change the ```sparse.py```

 thanks for the reminder)
